app/external_doc.rb

```diff
@@ -1,13 +1,121 @@
+require 'html/pipeline'
+require 'uri'
+
 class ExternalDoc
-  def self.fetch(url_to_markdown)
-    markdown = HTTP.get(url_to_markdown)
+  def self.fetch(repository:, path:)
+    contents = HTTP.get(
+      "https://raw.githubusercontent.com/#{repository}/master/#{path}",
+    )
+
+    context = {
+      base_url: URI.join(
+        'https://github.com',
+        "#{repository}/blob/master/",
+      ),
+
+      image_base_url: URI.join(
+        'https://raw.githubusercontent.com',
+        "#{repository}/master/",
+      ),
+    }
+
+    context[:subpage_url] =
+      URI.join(context[:base_url], File.join('.', File.dirname(path), '/'))
+
+    context[:image_subpage_url] =
+      URI.join(context[:image_base_url], File.join('.', File.dirname(path), '/'))
+
+    filters = [
+      HTML::Pipeline::MarkdownFilter,
+      HTML::Pipeline::AbsoluteSourceFilter,
+      PrimaryHeadingFilter,
+      TableOfContentsFilter,
+      AbsoluteLinkFilter,
+      MarkdownLinkFilter,
+    ]
+
+    HTML::Pipeline
+      .new(filters)
+      .to_html(contents, context)
```

Comment:
I'm probably wrong, but could this thing output markdown so that middleman will do the markdown -> HTML?